src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -0,0 +1,144 @@
+     * @throws ConnectException
+     */
+    @Override
+    public void publishBatch(List<SinkRecord> records) throws ConnectException {
+
+        if(!records.isEmpty()) {
+            Map<String, List<Message>> groupedMessages = groupMessagesByChannel(records);
+
+            if(groupedMessages != null && !groupedMessages.isEmpty())
+            {
+                groupedMessages.forEach((key, value) -> {
+                    logger.debug("Ably BATCH call -Thread(" + Thread.currentThread().getName() + ")" + "Num Records:" + value.size());
+                    try {
+                        sendBatches(new BatchSpec(Set.of(key), value));
+                    } catch (AblyException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Function to group the Ably Message objects by channel Name.
+     * @param sinkRecords
+     * @return
+     */
+    public Map<String, List<Message>> groupMessagesByChannel(List<SinkRecord> sinkRecords) {
+
+        HashMap<String, List<Message>> channelNameToMessagesMap = new HashMap();
+
+        for(SinkRecord record: sinkRecords) {
+            try {
+                if (!shouldSkip(record)) {
+                    final String channelName = channelSinkMapping.getChannelName(record);
+                    final Message message = messageSinkMapping.getMessage(record);
+
+                    List<Message> messages;
+                    if (channelNameToMessagesMap.containsKey(channelName)) {
+                        // just retrieve the list and add to it.
+                        messages = channelNameToMessagesMap.get(channelName);
+                    } else {
+                        // Create a new Set.
+                        messages = new ArrayList<>();
+                    }
+                    messages.add(message);
+                    channelNameToMessagesMap.put(channelName, messages);
+                }
+            } catch(Exception e) {
+                logger.error("Error transforming sink record", e);
+            }
+        }
+
+        return channelNameToMessagesMap;
+    }
+
+    protected boolean shouldSkip(SinkRecord record) {
+        final boolean skipOnKeyAbsence = connectorConfig.getBoolean(SKIP_ON_KEY_ABSENCE);
+
+        if (skipOnKeyAbsence) {
+            final String messageConfig = connectorConfig.getString(MESSAGE_CONFIG);
+            final String channelConfig = connectorConfig.getString(CHANNEL_CONFIG);
+            final ConfigValueEvaluator.Result messageResult = configValueEvaluator.evaluate(record, messageConfig, true);
+            final ConfigValueEvaluator.Result channelResult = configValueEvaluator.evaluate(record, channelConfig, true);
+
+            if (messageResult.shouldSkip() || channelResult.shouldSkip()) {
+                logger.warn("Skipping record as record key is not available in a record where the config for either" +
```

Comment:
This could be a really verbose log entry at the `WARN` level in some perfectly legit use cases. If I had a topic with 10k/sec records, and I'd like to forward 100/sec of those to Ably based on a particular key, I'd explicitly opt in to skipping records missing that key by using this config, but see an awful lot of log noise at a high severity level here.

Reply:
changed it to debug.